### PR TITLE
[SCM] Create all directories needed to reach the requested path (close #3423)

### LINKED ISSUE #3423: scm.subfolder support only one level of subfolders (not foo/baz/bar)
conan v1.6.1, Windows 10, Python 3.6

I'm moving our codebase to conan packages and encountered following error. I have to specify not single directory in `scm.subfolder` attribute, but multiple ones:
```python
scm = {
        "type": "git",
        "subfolder": "foo/baz/bar",
        "url": "auto",
        "revision": "auto",
    }
``` 
The reason is that I shouldn't change paths in `#include` directives today and I will update paths in the future.

With such recipe conan fails to build missing packages, i.e.
`conan install . --build missing` fails with an strange error:
`FileNotFoundError: [WinError 3] The system cannot find the path specified: 'C:\\.conan2\\gk8g1phl\\1\\foo\\baz\bar`

I think that source of this problem is this line:
https://github.com/conan-io/conan/blob/develop/conans/client/tools/scm.py#L20

Could you call `os.makedirs` instead of `os.mkdir`?

- [*] I've read the [CONTRIBUTING guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [*] I've specified the Conan version, operating system version and any tool that can be relevant.
- [*] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

### PR DESCRIPTION
**Pull request by jgsogo**
- [x] Refer to the issue that supports this Pull Request. #3423
- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] Add test

When downloading a recipe with nested directories in `SCM::subfolder` full path to final directory need to be created (not just the last dir).

### CODE REVIEW

**Comment by lasote** [comment]
Could you add a test?

File: conans/test/utils/cpp_test_files.py [hunk 1]
```diff
@@ -360,6 +362,13 @@ def cpp_hello_conan_files(name="Hello", version="0.1", deps=None, language=0, st
     for i in range(use_additional_infos):
         res += info_tmp % ("EnvVar%d" % i, "UserVar%d" % i)
 
+    scm_placeholder = ""
+    if scm_dict:
+        scm_dict_as_text = ', '.join(["'{}': '{}'".format(k, v) for k,v in scm_dict.items()])
```
Comment (memsharded) [inline conans/test/utils/cpp_test_files.py]: why a bare ``str(scm_dict)`` is not enough?

Comment (jgsogo) [reply conans/test/utils/cpp_test_files.py]: uhm.... sometimes you need to go the hard way just because... I'll change it :S

**Comment by jgsogo** [comment]
Replaced by #3440

**Pull request closed without merging**